cloudflare/data_source_devices_test.go

```diff
@@ -0,0 +1,52 @@
+package cloudflare
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccCloudflareDevices(t *testing.T) {
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("data.cloudflare_devices.%s", rnd)
+	accountID := os.Getenv("CLOUDFLARE_ACCOUNT_ID")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareDevicesConfig(rnd, accountID),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudflareDevices(name),
+					resource.TestCheckResourceAttr(name, "devices.#", "20"),
```

Comment:
i think for this to work, your config will also need to create the devices as right now, the acceptance account has zero. you can add these in your test below like we do with zones

https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/data_source_zones_test.go#L228-L267